x/escrow/client/cli/query.go

```diff
@@ -1,7 +1,142 @@
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	cmd.AddCommand(
+		cmdBlocksRemaining(),
+	)
+
+	return cmd
+}
+
+func cmdBlocksRemaining() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "blocks-remaining",
+		Short: "Compute the number of blocks remaining for an ecrow account",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			owner, err := cmd.Flags().GetString("owner")
+			if err != nil {
+				return err
+			}
+
+			dseq, err := cmd.Flags().GetUint64("dseq")
+			if err != nil {
+				return err
+			}
+
+			marketClient := marketTypes.NewQueryClient(clientCtx)
+			ctx := context.Background()
+
+			// Fetch leases matching owner & dseq
+			leaseRequest := marketTypes.QueryLeasesRequest{
+				Filters: marketTypes.LeaseFilters{
+					Owner:    owner,
+					DSeq:     dseq,
+					GSeq:     0,
+					OSeq:     0,
+					Provider: "",
+					State:    "active",
+				},
+				Pagination: nil,
+			}
+
+			leasesResponse, err := marketClient.Leases(ctx, &leaseRequest)
+			if err != nil {
+				return err
+			}
+
+			leases := make([]marketTypes.Lease, 0)
+			for _, lease := range leasesResponse.Leases {
+				leases = append(leases, lease.Lease)
+			}
+
+			// Fetch the balance of the escrow account
+			deploymentClient := deploymentTypes.NewQueryClient(clientCtx)
+			// fmt.Printf("length of leases, %d \n", len(leases))
+			outputSlice := make([]interface{}, 0)
+			for _, lease := range leases {
+				//  Fetch the time of last settlement
+				res, err := deploymentClient.Deployment(ctx, &deploymentTypes.QueryDeploymentRequest{
+					ID: lease.LeaseID.DeploymentID(),
+				})
+				if err != nil {
+					return err
+				}
+				amount := lease.Price.Amount
+				balance := res.EscrowAccount.Balance.Amount
+				settledAt := res.EscrowAccount.SettledAt
+				blockchainHeight, err := cli.CurrentBlockHeight(clientCtx)
+				if err != nil {
+					return err
+				}
+				balanceRemain := balance.Int64() - ((int64(blockchainHeight) - settledAt) * (amount.Int64()))
+				blocksRemain := balanceRemain / amount.Int64()
+				blocksPerDay := 86400 / 6.5
+				daysRemain := blocksRemain / int64(blocksPerDay)
+
+				output := struct {
+					BalanceRemain int64 `json:"balance_remaining" yaml:"balance_remaining"`
+					BlocksRemain  int64 `json:"blocks_remaining" yaml:"blocks_remaining"`
+					DaysRemain    int64 `json:"days_remaining" yaml:"days_remaining"`
+				}{
+					BalanceRemain: balanceRemain,
+					BlocksRemain:  blocksRemain,
+					DaysRemain:    daysRemain,
+				}
+				outputSlice = append(outputSlice, output)
+
+			}
+			outputType, err := cmd.Flags().GetString("output")
+			if outputType == "json" {
+				data, err := json.MarshalIndent(outputSlice, " ", "\t")
+				if err != nil {
+					return err
+				}
+				clientCtx.PrintBytes(data)
```

Comment:
There is a slight optimization here, you can declare the code like this and just set the values for data & err inside the the if / else and then use them outside the statement

```
var data []byte
if outputType =="json" {
  data, err = json.MarshalIndent
} else {
  data, err = yaml.Marshal
}

// Check for an error serializing
if err != nil {
  return err
}


//Write out the result
clientCtx.PrintBytes(data)
```